operators/channel_offset.py

```diff
@@ -79,35 +77,61 @@ class POWER_SEQUENCER_OT_channel_offset(bpy.types.Operator):
         description="Trim strips to make space in the target channel",
         default=False,
     )
+    keep_selection_offset: bpy.props.BoolProperty(
+        name="Keep selection offset",
+        description="The selected strips preserve their relative positions",
+        default=True,
+    )
 
     @classmethod
     def poll(cls, context):
         return context.selected_sequences
 
     def execute(self, context):
-        selection = [s for s in context.selected_sequences if not s.lock and not (self.direction == "down" and s.channel == 1) and not (self.direction == "up" and s.channel == 32)]
+        
+        max_channel = 32
+        min_channel = 1
+
+        if self.direction == "up":
+            movement = + 1
+            limit_channel = max_channel
+            movement_to_limit = min
+
+        if self.direction == "down":
+            movement = - 1
+            limit_channel = min_channel
+            movement_to_limit = max
+
+        selection = [s for s in context.selected_sequences if not s.lock]
+
         if not selection:
             return {"FINISHED"}
+        
+        sequences = sorted(selection, key=attrgetter("channel", "frame_final_start"))
+        if self.direction == "up":
+            sequences = [s for s in reversed(sequences)]
 
-        selection_blocks = slice_selection(context, selection)
-        for block in selection_blocks:
-            sequences = sorted(block, key=attrgetter("channel", "frame_final_start"))
-            frame_start, frame_end = get_frame_range(sequences)
-            channel_start, channel_end = get_channel_range(sequences)
-
-            if self.trim_target_channel:
-                to_delete, to_trim = find_strips_in_range(frame_start, frame_end, context.sequences)
-                channel_trim = (
-                    min(32, channel_end + 1) if self.direction == "up" else max(1, channel_start - 1)
-                )
-                to_trim = [s for s in to_trim if s.channel == channel_trim]
-                to_delete = [s for s in to_delete if s.channel == channel_trim]
-                trim_strips(context, frame_start, frame_end, to_trim, to_delete)
+        head = sequences[0]
+        if self.keep_selection_offset == False or (not head.channel == limit_channel and self.keep_selection_offset == True):
```

Comment:
You don't check for boolean equality in Python. You could write `if boolean_value is False`, which is more precise than `== False`, because booleans are numeric in Python (`False == 0` and `True == 1`). But you can omit the `==` or `is` altogether:

`if not self.keep_selection_offset`